pkg/apis/core/validation/validation.go

```diff
@@ -3899,45 +3904,90 @@ func validateSysctls(sysctls []core.Sysctl, fldPath *field.Path) field.ErrorList
+	return allErrs
+}
 
-		allErrs = append(allErrs, validateSeccompProfileField(securityContext.SeccompProfile, fldPath.Child("seccompProfile"))...)
+func validatePodWindowsSecurityContext(securityContext *core.PodSecurityContext, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	// OS based podSecurityContext validation
+	// There is some naming overlap between Windows and Linux Security Contexts but all the Windows Specific options
+	// are set via securityContext.WindowsOptions which we validate below
+	// TODO: Think if we need to relax this restriction or some of the restrictions
+	switch {
+	case securityContext.SELinuxOptions != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("SELinuxOptions"), "SELinux options cannot be set for Windows pods"))
+	case securityContext.HostPID:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("HostPID"), "HostPID cannot be set for Windows pods"))
+	case securityContext.HostIPC:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("HostIPC"), "HostIPC cannot be set for Windows pods"))
+	case securityContext.SeccompProfile != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("SeccompProfile"), "SeccompProfile cannot be set for Windows pods"))
+	case securityContext.FSGroup != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("FSGroup"), "FSGroup cannot be set for Windows pods"))
+	case securityContext.FSGroupChangePolicy != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("FSGroupChangePolicy"), "FSGroupChangePolicy cannot be set for Windows pods"))
+	case len(securityContext.Sysctls) > 0:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("Sysctls"), "Sysctls cannot be set for Windows pods"))
+	case securityContext.ShareProcessNamespace != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("ShareProcessNamespace"), "ShareProcessNamespace cannot be set for Windows pods"))
+	case securityContext.RunAsUser != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("RunAsUser"), "RunAsUser cannot be set for Windows pods"))
+	case securityContext.RunAsGroup != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("RunAsGroup"), "RunAsGroup cannot be set for Windows pods"))
+	case securityContext.SupplementalGroups != nil:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("SupplementalGroups"), "SupplementalGroups cannot be set for Windows pods"))
+	}
+	return allErrs
+}
+
+// ValidatePodSecurityContext test that the specified PodSecurityContext has valid data.
+func ValidatePodSecurityContext(securityContext *core.PodSecurityContext, spec *core.PodSpec, specPath, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	os := spec.OS
+	if securityContext != nil {
+		if os == nil || (len(os.Name) == 0 || os.Name == core.Linux) {
+			allErrs = append(allErrs, validatePodLinuxSecurityContext(securityContext, spec.Containers, specPath, fldPath)...)
```

Comment:
calling validatePodLinuxSecurityContext if os == nil is really confusing, and potentially wrong if it starts forbidding windows-specific fields... I'd recommend an approach more like https://github.com/kubernetes/kubernetes/pull/104693/commits/8753e07184294523d430998580f3f20b13223ebe#r719497809